libpod/events/logfile.go

```diff
@@ -224,9 +300,25 @@ func truncate(filePath string) error {
 			return err
 		}
 	}
+
+	if err := writeRotateEvent(tmp, filePath, true); err != nil {
+		return fmt.Errorf("writing rotation event begin marker: %w", err)
+	}
 	if _, err := reader.WriteTo(tmp); err != nil {
 		return fmt.Errorf("writing truncated contents: %w", err)
 	}
+	if _, err := tmp.WriteString(""); err != nil {
+		return err
+	}
+	if err := writeRotateEvent(tmp, filePath, false); err != nil {
+		return fmt.Errorf("writing rotation event end marker: %w", err)
+	}
+	if err := tmp.Close(); err != nil {
+		return err
+	}
+	if err := orig.Close(); err != nil {
+		return err
+	}
```

Comment:
Both will be already closed by the defer statement. This will cause it to be closed twice.